src/include/duckdb/main/client_properties.hpp

```diff
@@ -15,13 +15,15 @@ enum ArrowOffsetSize { REGULAR, LARGE };
 
 //! A set of properties from the client context that can be used to interpret the query result
 struct ClientProperties {
-	ClientProperties(string time_zone_p, ArrowOffsetSize arrow_offset_size_p, bool produce_arrow_string_view_p)
+	ClientProperties(string time_zone_p, ArrowOffsetSize arrow_offset_size_p, bool arrow_use_list_view_p,
+	                 bool produce_arrow_string_view_p)
 	    : time_zone(std::move(time_zone_p)), arrow_offset_size(arrow_offset_size_p),
-	      produce_arrow_string_view(produce_arrow_string_view_p) {
+	      arrow_use_list_view(arrow_use_list_view_p), produce_arrow_string_view(produce_arrow_string_view_p) {
 	}
 	ClientProperties() {};
 	string time_zone = "UTC";
 	ArrowOffsetSize arrow_offset_size = ArrowOffsetSize::REGULAR;
+	bool arrow_use_list_view = false;
```

Comment:
Given that this is a global setting, I doubt there will ever be a point changing it to `true` doesn't break a lot of stuff. 

How much reliance on this global there is instead of inferring from context which type should be produced?

A somewhat radical proposal: make it default to `true` to incentivize the ecosystem to move to list-view (more flexible and Velox-default). The disadvantage is that things would more often break in the near future and you would have to remind people to set it to `false` to "unbreak" things, but in the long term we end-up with a better default.

Reply:
Another idea: could it be a tri-state? `true`, `false`, and `auto`? With `auto` giving you more freedom to decide what type to use. e.g. if you know you're exporting to Velox (somehwow) and the setting is `auto`, you pick list-view because Velox will be able to import that in a zero-copy way. Even if you can't infer what type you can use, `auto` would allow you to change the default in the future without breaking the promise to any user that sets it to `true` or `false` explicitly.

Reply:
I'll discuss it internally
I'm not sure how keen we are on changing the default behavior.

I agree that it's more flexible, but when we produce list views we make no use of this flexibility

Reply:
For now we'll keep it as `false` by default, to not break existing behavior.
We can revisit the issue when we've launched 1.0 and ListViewArray is part of a release 👍

Reply:
> We can revisit the issue when we've launched 1.0 and ListViewArray is part of a release 👍

Understandable. But to keep this an open option, you would have to differentiate between users explicitly setting it to either true or false versus not setting anything at all, hence my tri-state suggestion instead of just a boolean.

Reply:
The option has a default value, which is false

All of our options can be [reset](https://duckdb.org/docs/sql/statements/set#reset) to their default value

So in that sense it is pretty much a tri-state already no?
Where "auto" is just not setting it explicitly

Reply:
There are 3 types of users:

1. those that write `set option = true`
2. those that write `set option = false`
3. those that don't set the option anywhere

If you change the de facto default on the 1.0 release, you will disappoint type 1 and type 2 users who intentionally wrote the `set option` line. You can only really change the de facto default internally to type 3 users. And even that is a little iffy. You would do that only if the risk of breakage was very low.

> All of our options can be [reset](https://duckdb.org/docs/sql/statements/set#reset) to their default value

And the default value is documented, so users that type `RESET` are type 1 or type 2 users depending on what is documented.

> So in that sense it is pretty much a tri-state already no?
> Where "auto" is just not setting it explicitly

Not really, because your code can't distinguish between explicity set (or explicitlye reset) versus not set at all. If it could also be NULL (like other options), you would have that ability and more freedom in format choice for these users.